.travis.yml

```diff
@@ -35,61 +29,91 @@ matrix:
   include:
     - os: linux
       sudo: required
-      dist: trusty
-      language: cpp
+      services: docker
+      env: RUNMODE=docker
+    - os: linux
+      sudo: required
       compiler: gcc
+      env: RUNMODE=travis
+      addons:
+        # If using C++11/14 see http://genbattle.bitbucket.org/blog/2016/01/17/c++-travis-ci/
+        apt:
+          packages: [ openssh-client, openssh-server, liblapack-pic, liblapack-dev ]
     - os: osx
       #osx_image: xcode7.3
-      language: cpp
       compiler: clang
+      env: RUNMODE=travis
 
 before_install:
-  # Configure ssh
-  - ssh-keygen -t rsa -f ~/.ssh/id_rsa -N ""
-  - chmod 700 ~/.ssh/
-  - cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
-  - chmod 600 ~/.ssh/authorized_keys
-  - ssh-keyscan -t rsa localhost >> ~/.ssh/known_hosts
-  - ssh-keyscan -t rsa 127.0.0.1 >> ~/.ssh/known_hosts
-  - chmod 600 ~/.ssh/known_hosts
-  - ls -al ~/.ssh/
-  - ssh localhost ls
-  - ssh 127.0.0.1 ls
-  # Install minimal conda
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
-      wget https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh -O miniconda.sh;
+  - |
+    if [[ "$RUNMODE" == "docker" ]] ; then
+      # Configure docker
+      DOCKER=solvcon/solvcon_build
+      docker pull $DOCKER
+
+      containerid=$(docker run --detach --tty \
```

Comment:
The docker job runs faster than travis native jobs, because the docker image contains all the dependencies and doesn't need conda.